File: Python/speeding.py
'''
 *
 *	Problem: "Speeding"
 *
'''

numCases = int(input())

timeSpeed = input().split()
prevTime = int(timeSpeed[0])
prevSpeed = int(timeSpeed[1])

maxSpeed = 0
for case in range( (numCases - 1) ):
	timeSpeed = input().split(" ")
	currTime = int(timeSpeed[0])
	currSpeed = int(timeSpeed[1])
	
	tempMaxSpeed = (currSpeed - prevSpeed) // (currTime - prevTime)
	if ( (tempMaxSpeed) > maxSpeed ):
		maxSpeed = tempMaxSpeed
	
	prevTime = currTime
	prevSpeed = currSpeed
	
print(maxSpeed)

File: Java/speeding.java
/*
 *	Problem: "Speeding"
 */

import java.util.Scanner;

public class speeding {

	/*-----main-----*/
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int numCases = sc.nextInt();

		int prevTime = sc.nextInt();
		int prevSpeed = sc.nextInt();
		int maxSpeed = 0;
		
		while (--numCases > 0) {

		int currTime = sc.nextInt();
		int currSpeed = sc.nextInt();
			
		int tempMaxSpeed = (currSpeed - prevSpeed) / (currTime - prevTime);
		if (tempMaxSpeed > maxSpeed)
			maxSpeed =  tempMaxSpeed;
		prevTime = currTime;
		prevSpeed = currSpeed;
		}
		
		System.out.println(maxSpeed);
	} // main
	
}